Checkout: Thank You: Use `Gridicon` instead of noticons

drewblaisdell opened the pull request:
This PR:

- Updates the icons on `CheckoutThankYou` per @breezyskies' spreadsheet.
- Updates `PurchaseDetail` to accept an `icon` prop that is used to mount a `Gridicon`, instead of using noticons in CSS.

**Testing**

We should assert that the icons on the thank you page render correctly after purchasing the following items:

```
business-plan
chargeback
domain-mapping
domain-registration
google-apps
jetpack-business-plan
jetpack-premium-plan
premium-plan
site-redirect
```

- [x] Code review
- [ ] Product review

drewblaisdell commented:
We can't test the Jetpack-specific icons here until [this PR](https://github.com/Automattic/wp-calypso/pull/3617) is merged.

scruffian commented:
Some issues I noticed:
1. The back to my site button looks broken:
<img width="840" alt="screen shot 2016-02-29 at 13 18 43" src="https://cloud.githubusercontent.com/assets/275961/13395493/00c85ad4-dee7-11e5-828f-cd0c0ce1d08b.png">

2. The icons are grey on a white background, not white on a grey background:
<img width="212" alt="screen shot 2016-02-29 at 13 19 11" src="https://cloud.githubusercontent.com/assets/275961/13395507/182a9cfa-dee7-11e5-9777-38a42ae4d658.png">

scruffian commented:
I updated ec7dcf4 to fix the bugs I noticed above ^

@breezyskies @mikeshelton1503  this could do with a design review.

mtias commented on client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx:
```diff
@@ -8,22 +8,24 @@ import React from 'react';
 
 	return (
 		<div className={ classes }>
+			{ icon && <div className="gridicon-wrapper"><Gridicon icon={ icon } size={ 72 } /></div> }
```
This class should be namespaced to the component: `checkout-thank-you__gridicon-wrapper`.

mtias commented on client/my-sites/upgrades/checkout-thank-you/style.scss:
```diff
@@ -74,86 +74,11 @@
 		text-align: left;
 	}
 
-	&::before {
```
Great to see this :)

mtias commented:
This is cool. I think `purchase-detail.jsx` could be its own component outside of `checkout-thank-you` by the way. Something like `<PlanFeatureReminder>` or something like that.

breezyskies commented:
A couple sizing tweaks for the icons:

Desktop:
- Icon size: 48x48
- `padding: 16px`

Mobile (<660px):
- Icon size: 32x32
- `padding: 16px`

fabianapsimoes commented:
Are we not showing the "Back to my site" label for mobile? This is how it looks now:

![screen shot 2016-02-29 at 17 35 58](https://cloud.githubusercontent.com/assets/1248436/13401059/e9e4beba-df0a-11e5-9fcd-fce0347ca68c.png)

fabianapsimoes commented:
Some gridicons look quite misaligned still. For example:

![screenshot-calypso localhost 3000 2016-02-29 17-45-10](https://cloud.githubusercontent.com/assets/1248436/13401547/d4526aa0-df0c-11e5-8083-9bd3527e62d4.png)

scruffian commented:
> Icon size: 32x32

@breezyskies - is it ok to show Gridicons at 32px? They generally ask that we show them at a multiple of 24...

fabianapsimoes commented:
Seems we're missing an icon here, for Google Apps purchases:
![screenshot-calypso localhost 3000 2016-02-29 17-54-43](https://cloud.githubusercontent.com/assets/1248436/13401803/eb9e4db8-df0d-11e5-89fb-9a8b5dfd5509.png)

breezyskies commented:
> is it ok to show Gridicons at 32px? They generally ask that we show them at a multiple of 24...

Let's go with 36px, per [this documentation](https://wpcalypso.wordpress.com/devdocs/client/components/gridicon/README.md?term=gridicons).

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx:
```diff
@@ -8,22 +8,24 @@ import React from 'react';
 
 	return (
 		<div className={ classes }>
+			{ icon && <div className="checkout-thank-you__gridicon-wrapper"><Gridicon icon={ icon } size={ 72 } /></div> }
```
@scruffian We should namespace this class under the subcomponent namespace, like: `checkout-thank-you__purchase-detail-gridicon-wrapper`.

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/style.scss:
```diff
@@ -183,4 +108,30 @@
 		}
 	}
+
+	.checkout-thank-you__gridicon-wrapper {
```
@scruffian We don't need to include this selector inside the `.checkout-thank-you__purchase-detail` ruleset.

drewblaisdell commented:
I made some changes:

- Updated the icon sizes per @breezyskies' note.
- Fixed the icon for Google Apps (the icon slug was `email` instead of `mail`).
- Moved `PurchaseDetail` to `client/components/` per @mtias' recommendation, and updated the CSS accordingly. I kept the name `PurchaseDetail`, instead of the suggested `PlanFeatureReminder`, because it is simpler and because most of our purchase items are not plans.

> Are we not showing the "Back to my site" label for mobile? This is how it looks now:

Not related to this PR specifically, but the default behavior of `HeaderCake` is to hide text from the `backText` prop ("Back to my site", in this case) on mobile viewports.

> Some gridicons look quite misaligned still. For example:

I'm not seeing any more misaligned Gridicons after the switch to 48px/36px icons ([clock screenshot](https://cloudup.com/cdIhjDOuB0g)) but please, let me know if you still notice them.

breezyskies commented:
Icon sizes look good :+1:

scruffian commented:
Code :+1:

fabianapsimoes commented:
Product :+1:
